contract-service/src/main/java/com/example/contractservice/contract/service/ContractService.java

```diff
@@ -53,6 +57,32 @@ public class ContractService {
     @Value("${module.member.application.path.member-info}")
     private String memberInfoUrl;
 
+    public List<ContractBriefWithNicknameResponse> getBriefInfos(List<String> codes) {
+        // 코드를 기반으로 모든 ContractEntity를 한 번에 조회
+        List<ContractEntity> contractEntities = contractRepository.findAllByCodes(codes);
+
+        if (contractEntities.isEmpty()) { // 없다면 조기 종료로 네트워크 통신 방지
+            return Collections.emptyList();
+        }
+
+        // 계약 목록에서 요청자(requestor)와 계약자(contractor)의 code를 모두 수집 (중복 제거)
+        Set<String> memberCodes = contractEntities.stream()
+                .flatMap(entity -> Stream.of(entity.getRequestorCode(), entity.getContractorCode()))
+                .collect(Collectors.toSet());
+
+        // member 모듈로부터 정보 가져오기
+        URI memberInfoUri = createMemberInfoUrl(memberCodes.stream().toList());
+        List<MemberInfo> memberInfos = Optional.ofNullable(restTemplate.getForObject(memberInfoUri, MemberInfoResponse.class))
+                .orElseThrow(() -> new ContractException(INVALID_MEMBER))
+                .members();
+        Map<String, String> membersByCode = memberInfos.stream()
+                .collect(Collectors.toMap(MemberInfo::code, MemberInfo::name)); // code별로 info 분류
```

Comment:
현재 로직에서는 `MemberInfo::code`의 중복이 제거되어 있어서 문제가 발생하지는 않겠지만, 혹시 중복될 수 있는 상황에서는 병합 함수를 사용하는 게 좋겠네요! 


```suggestion
        Map<String, String> membersByCode = memberInfos.stream()
                .collect(Collectors.toMap(MemberInfo::code, MemberInfo::name, (a, b) -> a))); // code별로 info 분류
```

Reply:
좋은 의견과 코드 제안 감사드립니다.